pkg/networkutils/network.go

```diff
@@ -489,22 +488,31 @@ func (n *linuxNetwork) buildIptablesSNATRules(vpcCIDRs []string, primaryAddr *ne
 		}
 	}
 
-	lastChain := chains[len(chains)-1]
-	iptableRules = append(iptableRules, iptablesRule{
-		name:        "last SNAT rule for non-VPC outbound traffic",
-		shouldExist: !n.useExternalSNAT,
-		table:       "nat",
-		chain:       lastChain,
-		rule:        snatRule,
-	})
-
 	snatStaleRules, err := computeStaleIptablesRules(ipt, "nat", "AWS-SNAT-CHAIN", iptableRules, chains)
 	if err != nil {
 		return []iptablesRule{}, err
 	}
 
 	iptableRules = append(iptableRules, snatStaleRules...)
 
+	// Force delete the SNAT rule as we want this to be the last rule of the AWS-SNAT-CHAIN-0 chain
```

Comment:
Could you expand on this? Why do we have to delete this rule? Is this from the old chain (before this change)?

Reply:
Say we had the rules:
-d <CIDR1> -j RETURN
-d <CIDR2> -j RETURN
SNAT Rule

If we were to add a CIDR, we want to ensure the SNAT Rule is after that CIDR rule. If we didn't force delete this SNAT rule and readd it, it could potentially become:
-d <CIDR1> -j RETURN
-d <CIDR2> -j RETURN
SNAT Rule
-d <CIDR3> -j RETURN
This is unwanted behavior as the CIDR3 rule won't even be checked.

Reply:
Got it, this is order within the chain, thanks for explaining. I suppose this can have an impact to existing traffic (while the rules are being updated), so we should think if there is any way to delete the rule last